### PR TITLE
jQuery: Add article aliases (redirects) -- Example

### LINKED ISSUE #613: jQuery: Add article aliases (redirects) -- Example
### Description

Add aliases appending the word "example" to all article titles.

### What to do
For each article line in the [articles.txt file](https://github.com/duckduckgo/zeroclickinfo-fathead/blob/master/lib/fathead/jquery/articles.txt), create aliases and store them in a new redirects.txt file, in the same directory.

Each line of the redirects.txt file should be structured as follows: alias,original line.
E.g: `ajax example,jQuery.ajax()`

The aliases should have:
- The word "jQuery" removed
- Symbols such as dots replaced with spaces
- Parentheses removed
- CamelCase words split into different words (eg "createClass" would become "create class")

This change will allow the Instant Answer to trigger for more searches.

## Get Started
- [ ] 1) Claim this issue by commenting below
- [ ] 2) Review our [Contributing Guide](https://github.com/duckduckgo/zeroclickinfo-fathead/blob/master/CONTRIBUTING.md)
- [ ] 3) [Set up your development environment](https://docs.duckduckhack.com/welcome/setup-dev-environment.html), and fork this repository
- [ ] 4) Create a Pull Request

## Resources
- Join [DuckDuckHack Slack](https://quackslack.herokuapp.com/) to ask questions
- Join the [DuckDuckHack Forum](https://forum.duckduckhack.com/) to discuss project planning and Instant Answer metrics
- Read the [DuckDuckHack Documentation](https://docs.duckduckhack.com/) for technical help

------
IA Page:  https://duck.co/ia/view/jquery

### PR DESCRIPTION
**Pull request by Caldwerl**
## Description of new Instant Answer, or changes
For each article line in the articles.txt file, create aliases and store them in a new redirects.txt file, in the same directory.

Each line of the redirects.txt file is structured as follows: alias,original line.
E.g: ajax example,jQuery.ajax()

The aliases have:
- The word "jQuery" removed
- Symbols such as dots replaced with spaces
- Parentheses removed
- CamelCase words split into different words (eg "createClass" would become "create class")

This change will allow the Instant Answer to trigger for more searches.

There is an edge case with jQuery(), it is currently set to skip writing that line, but it can be changed to include an alias for that as well.

## Related Issues and Discussions
Fixes #613

## People to notify
@bfmags 

## Testing & Review
To be completed by Language Leader (or DDG Staff)

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page**
- [ ] Instant Answer page is correctly filled out and contains:
    - One topic for the Search Space Language (Java, Python, Scala, Ruby, etc.)
    - One topic from: Reference, Help, Libraries, Tools
        - Documentation Fatheads are considered "Reference"
    - Description, Data source, and 2+ example queries
    - Perl Module (e.g. "DDG::Fathead::PerlDoc" -- we only need a name, not an actual file)
    - Source Name (for "More at <source_name>" link)
    - Source Domain (must contain http:// or https:// -- can be the same as Data Source)
    - Source Info (used as Subtitle for each Article -- usually matches the IA Name)
    - 'Skip Abstract' is checked off
    - Source ID (ping @moollaza to assign one, once Fathead is ready for Beta deploy)
- [ ] Fathead Tests are passing (run `$ duckpan test <fathead_id>`)
    - Tester should report any failures

**Code**
- [ ] Uniformly indented, well commented
- [ ] Fetch.sh and Parse.sh run without errors
- [ ] Output contains no blank lines, or multi-line entries

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/jquery

### CODE REVIEW

**Comment by daxtheduck** [comment]
@Caldwerl Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/fathead/jquery/article-alias.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+var rl = readline.createInterface(fs.createReadStream(filePath + inputFileName), fs.createWriteStream(filePath + outputFileName,{'flags': 'a'}));
+
+function parseInput(inputString) {
+    var outputString = inputString;
+
+    function upperToLowerSpace(match) {
+        return ' ' + match.toLowerCase();
+    }
+
+    outputString = outputString.replace('jQuery', '').replace(/[\(\)]/g, '').replace(/[\.\:\-]/g, ' ').replace(/([A-Z]+)/g, upperToLowerSpace).replace(/\ \ /g, ' ').trim();
```
Comment (bfmags) [inline lib/fathead/jquery/article-alias.js]: @Caldwerl This looks great! 👍 

One thing though, can you add another line for each camel case article, besides the split words, with the original title, like this example:
`remove class example,.removeClass()`
`removeClass example,.removeClass()`

Also, committing the redirects.txt will trigger the deploy of the fathead to beta.duckduckgo.com, so we can test it :)

Comment (Caldwerl) [inline lib/fathead/jquery/article-alias.js]: @bfmags 
Sure thing, I will get on it when I get home today.
Thanks.

**Comment by bfmags** [comment]
Awesome! Thanks! 🎉 😄

**Pull request merged**